gtk: fix duplicate signal handlers

jcollie opened the pull request:
Signal handlers are connected to surface objects in two spots - when a tab is added to a page and when the split tree changes. This resulted in duplicate signal handlers being added for each surface. This was most noticeable when copying the selection to the clipboard - you would see two "Copied to clipboard" toasts. Ensure that there is only one signal handler by removing any old ones before adding the new ones.

jcollie committed:
gtk: fix duplicate signal handlers

mitchellh reviewed (changes requested):
Question: This is a simple change, but would it be more robust to change `connectSurfaceHandlers` to `setSurfaceHandlers` and change `tree` to an optional. If it is null then we just disconnect. Then we can replace all usage everywhere with the one call. Thought?

jcollie committed:
gtk: improve signal handler management

Instead of making two separate passes over the surfaces in a split tree
to manage signal handlers, do it in one pass.

jcollie commented:
Ok, I've updated the patch so that it only makes one pass through the surfaces rather than two.
